9.5/root/usr/share/container-scripts/postgresql/README.md

```diff
@@ -96,3 +96,38 @@ values stored in the variables and the actual passwords. Whenever a database
 container starts it will reset the passwords to the values stored in the
 environment variables.
 
+
+Upgrading database
+------------------
+
+** Warning! Please, before you decide to do the data directory upgrade, always
+ensure that you've carefully backed up all your data and that you are OK with
+potential manual rollback! **
+
+This image supports automatic upgrade of data directory created by
+PostgreSQL server 9.4 (and only by this version) - provided by sclorg image.
+The upgrade process is designed so that you should be able to just switch from
+*image A* to *image B*, and set the `$POSTGRESQL_UPGRADE` variable
+appropriately to explicitly request the data transformation.
+
+The upgrade process is internally implemented by `pg_upgrade` binary, and for
+that purpose the container needs to contain two versions of PostgreSQL server
+(have a look at `man pg_upgrade` for more info).
+
+For the `pg_upgrade` process - and the new server version, we need to initialize
+a brand new data directory.  That's data directory is created automatically by
+container tooling under /var/lib/pgsql/data, which is usually external
+bind-mountpoint.  The `pg_upgrade` execution is then similar to dump&restore
+approach -- it starts both old and new PostgreSQL servers (within container) and
+"dumps" the old datadir while and at the same time it "restores" it into new
+datadir.  This operation requires a lot of data files copying, so you can decide
+what type of upgrade you'll do by setting `$POSTGRESQL_UPGRADE` appropriately:
+
+|    Variable value  |    Description                                 |
+| :----------------- | ---------------------------------------------- |
+|  `copy`            | The data files are copied from old datadir to new datadir.  This option has low risk of data loss in case of some upgrade failure. |
```

Comment:
If we remove that option, we remove the only significant benefit compared to #165 -> "faster" upgrade.